gcore/gdaldriver.cpp

```diff
@@ -1478,6 +1478,23 @@ bool GDALDriver::CanVectorTranslateFrom(
     return bRet;
 }
 
+bool GDALDriver::HasOpenOption(const char *pszOpenOptionName) const
+{
+    if (pszOpenOptionName == nullptr)
+        return false;
+
+    auto pszOOMd{const_cast<GDALDriver *>(this)->GetMetadataItem(
+        GDAL_DMD_OPENOPTIONLIST)};
+    if (pszOOMd == nullptr)
+        return false;
+
+    // Const cast is safe here since we are only reading the metadata
+    const std::string osOpenOptions{pszOOMd};
+    const std::string osKey =
+        "<Option name='" + std::string(pszOpenOptionName) + "'";
```

Comment:
I guess that will work in practice, but that's a bit fragile. We could potentially have a driver using escaped double quote ``"<Option name=\"foo\" ....>"``. Or reordering attributes with the name not being first.
The clean solution would be to use CPLXMLParseString() and browse through the structure. I'm OK though with that simplified implementation, but perhaps add a code comment reflecting my observation.

Reply:
I thought that being us in control of the source XML it could be acceptable but better be robust if we can.